src/backend/distributed/deparser/deparse_view_stmts.c

```diff
@@ -92,3 +99,206 @@ AppendViewNameList(StringInfo buf, List *viewNamesList)
 		isFirstView = false;
 	}
 }
+
+
+/*
+ * DeparseAlterViewStmt deparses the given ALTER VIEW statement.
+ */
+char *
+DeparseAlterViewStmt(Node *node)
+{
+	AlterTableStmt *stmt = castNode(AlterTableStmt, node);
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	AppendAlterViewStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+static void
+AppendAlterViewStmt(StringInfo buf, AlterTableStmt *stmt)
+{
+	const char *identifier = quote_qualified_identifier(stmt->relation->schemaname,
+														stmt->relation->relname);
+
+	appendStringInfo(buf, "ALTER VIEW %s ", identifier);
+
+	AlterTableCmd *alterTableCmd = castNode(AlterTableCmd, lfirst(list_head(stmt->cmds)));
+	AppendAlterViewCmd(buf, alterTableCmd);
+
+	appendStringInfoString(buf, ";");
+}
+
+
+static void
+AppendAlterViewCmd(StringInfo buf, AlterTableCmd *alterTableCmd)
+{
+	switch (alterTableCmd->subtype)
+	{
+		case AT_ChangeOwner:
+		{
+			AppendAlterViewOwnerStmt(buf, alterTableCmd);
+			break;
+		}
+
+		case AT_SetRelOptions:
+		{
+			AppendAlterViewSetOptionsStmt(buf, alterTableCmd);
+			break;
+		}
+
+		case AT_ResetRelOptions:
+		{
+			AppendAlterViewResetOptionsStmt(buf, alterTableCmd);
+			break;
+		}
+
+		case AT_ColumnDefault:
+		{
```

Comment:
do we throw errors in these deparser code or earlier in general? This seems a little too late to throw the error?

Reply:
For example,
```SQL

 alter table "r1 owned schama"."r1 owned view 2" ADD COLUMN x int;
ERROR:  unsupported subtype for alter view command
DETAIL:  sub command type: 0

```
Should be as in PG:
```SQL
ERROR:  "r1 owned view 2" is not a table, composite type, or foreign table
```

Is that easy to achieve? I guess even if move deparse to post_process, that should be give the same errors as PG?

Reply:
Since ALTER VIEW command can handle only the cases above, I've removed throwing error from the default case to let PG to throw the right error locally.